docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,312 @@
+   "metadata": {},
+   "source": [
+    "### Get granule response and granule download URL with `get_granules` and `get_granule_link` for collection `M2I1NXASM`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "granule_response = get_granules(concept_id='C1256535511-EEDTEST',\n",
+    "                                cmr_env=CMR_UAT,\n",
+    "                                auth_header=auth_header)\n",
+    "link = get_granule_link(granule_response)\n",
+    "link"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Then download the granule locally with `download_granule`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Saves granule to current directory or\n",
+    "# out_directory=/usr/given/path\n",
+    "download_granule(link, auth_header=auth_header)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Instantiate a `VarInfoFromNetCDF4` object for a local NetCDF-4 file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate a VarInfoFromNetCDF4 object for a local NetCDF-4 file.\n",
+    "var_info = VarInfoFromNetCDF4('MERRA2_400.inst1_2d_asm_Nx.20220130.nc4',\n",
+    "                              short_name='M2I1NXASM')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of UMM-Var JSON records"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_dict = get_all_umm_var(var_info)\n",
+    "umm_var_dict"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish all the UMM-Var entries for M2I1NXASM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "publish_all_umm_var('C1256535511-EEDTEST',\n",
+    "                    umm_var_dict,\n",
+    "                    auth_header=auth_header,\n",
+    "                    cmr_env=CMR_UAT)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish a single UMM-Var entry"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_entry = get_umm_var(var_info, var_info.get_variable('/TROPPV'))\n",
```

Comment:
Suggestion: 
TROPPV_var = var_info.get_variable('/TROPPV')
if TROPPV_var:  # best to check for None first
    umm_var_entry = get_umm_var(var_info, TROPPV_var)
else
    print("/TROPPV not found in granule")

Reply:
Overall structure looks great.

Putting the old PEP8 hat on for a nit: checks against `None` should be: `if TROPPV_var is not None`, rather than just `if TROPPV_var`.

Reply:
√ - I bow to PEP8!